packages/gatsby/index.d.ts

```diff
@@ -1521,3 +1522,57 @@ export interface IPluginRefOptions {
 export interface PluginOptionsSchemaArgs {
   Joi: PluginOptionsSchemaJoi
 }
+
+/**
+ * Send body of response
+ */
+type Send<T> = (body: T) => void
+
+/**
+ * Gatsby API Function route response
+ */
+export type GatsbyAPIFunctionResponse<T = any> = ServerResponse & {
```

Comment:
Can this be an interface that extends ServerResponse, rather than an intersection?

Reply:
Could you tell me like I'm 5 why this is a good idea? I'm happy to do it but would also love to understand the reason.

Reply:
I guess extension is only useful if you want to override a type which we're not so `extends` is fine https://stackoverflow.com/a/52681859/182702

Reply:
https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections